+nosnoc/+time_freezing/cls_elastic.m

```diff
@@ -0,0 +1,74 @@
+function [pss_model] = cls_elastic(cls_model, opts)
+    dims = cls_model.dims;
+    pss_model = nosnoc.model.Pss();
+    pss_model.dims = dims;
+
+    % transfer all common properties to pss_model
+    common_props = properties('nosnoc.model.Base');
+    for ii=1:length(common_props)
+        prop = common_props{ii};
+        pss_model.(prop) = cls_model.(prop);
+    end
+    
+    dims.n_contacts = length(cls_model.f_c);
+    % check dimensions of contacts
+    if isempty(dims.n_dim_contact)
+        warning('nosnoc: Please n_dim_contact, dimension of tangent space at contact (1, 2 or 3)')
+        dims.n_dim_contact = 2;
+    end
+    
+    % quadrature state
+    dims.n_quad = 0;
+    f_quad = [];
+    if opts.time_freezing_quadrature_state
+        % define quadrature state
+        L = define_casadi_symbolic(opts.casadi_symbolic_mode,'L',1);
+        pss_model.lbx = [pss_model.lbx;-inf];
+        pss_model.ubx = [pss_model.ubx;inf];
+        pss_model.x = [pss_model.x;L];
+        pss_model.x0 = [pss_model.x0;0];
+        f_quad = pss_model.f_q;
+        pss_model.f_q = 0;
+        if ~isempty(pss_model.f_q_T)
+            pss_model.f_q_T  = pss_model.f_q_T + L;
+        else
+            pss_model.f_q_T = L;
+        end
+        dims.n_quad = 1;
+    end
+    
+    % uneven number of states = it is assumed that the clock state is defined.
+    t = define_casadi_symbolic(opts.casadi_symbolic_mode,'t',1);
+    % update lower and upper bounds of lbx and ubx
+    pss_model.lbx = [pss_model.lbx;-inf];
```

Comment:
we have natural lower bounds for the clock state, 
it must be nonnegative, but can put -1 for some relaxation
upper bound is also not much bigger than T (except in time optimal) but a suff large value, .e.g. 100 or 1000 might help convergence...

Reply:
Do we want to add options for these? I suspect the effects here would be mild, though the implementation is not too difficult so it may be worth it.
=